### DISCUSSION TITLE
how could it work with remotely fetched assets?

### DISCUSSION THREAD

**Discussion by pierroo** [dicebear/dicebear#371] [Q&A] [answered]
I am trying to build a react native avatar editor.
So far it works great to create an avatar with static / local assets.

However, how would it work if I wanted to push new assets / cosmetics remotely?
for example new hair, new eyes etc, without having to update the apk?

for example this is the current components/eyes.js, that is present locally, and contains 3 eyes variations:
![image](https://github.com/dicebear/dicebear/assets/17619412/494ead4f-c45b-4b99-b852-9c0d9d89ce13)

My designer creates awesome new eyes, adds them in figma as per your documentation, how could I append them to the existing dicebear editor, without having to re-run the figma export; npm install / npm run build => get the lib folder, and have to re-update the app/apk with this new styles folder?

I hope I am not asking beyond this library scope, which is already doing an amazing job;
but in terms of flexibility for further added assets I may not master how it works perfectly enough.

Thank you in advance!

**Comment by FlorianKoerner** [comment, marked as answer]
Maybe a fork of the [dicebear/api](https://github.com/dicebear/api) that only provides your avatar style. Then you could read the schema.json (e.g. https://api.dicebear.com/7.x/adventurer/schema.json) and provide the corresponding avatar options and the created avatar. But that would be an online-first approach.

**Reply by pierroo** [reply]
Thank you once again for your suggestion @FlorianKoerner ;
I had considered this approach, unfortunately from this schema.json I only get the list of variations names;

But how could I translate these variations name into an actual svg file for the front end?

My worries is having to manually re-export the figma export plugin each time I add / change a new asset, instead of having a way to "merge" / add one onto the existing list without breaking the dicebear structure;

I see you have created the dicebear editor here: https://github.com/dicebear/editor
somehow this is something like that I am trying to achieve, but listing accessories only on that bottom half, and have them remotely fetched

**Reply by FlorianKoerner** [reply]
My suggestion includes that the avatars are then also created server-side and not directly in the app.

At the same time, I think it shouldn't be a problem to update the APK regularly. Just don't edit the created folder for the avatar style so it can always be replaced with a new version.
